dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -820,13 +817,13 @@ public Workflow getWorkflowByPath(@ApiParam(hidden = true) @Parameter(hidden = t
         return workflow;
     }
 
-    private Class<? extends Workflow> getSubClass(String subclass) {
+    private Class<? extends Workflow> getSubClass(WorkflowSubClass subclass) {
         final Class<? extends Workflow> targetClass;
-        if (subclass.equals(SERVICE)) {
+        if (subclass.equals(WorkflowSubClass.SERVICE)) {
```

Comment:
You can use `==` with enums. The `equals` method also works, but the advantage of `==` is that you don't have to do a null check. Here you are assuming it's never null, which I think is OK, but it's confusing with the different ways the endpoint will get invoked, so I'm not sure. Probably OK, your call.